File: java/spring-boot/user/src/main/java/com/learnwy/user/service/impl/UserServiceImpl.java
package com.learnwy.user.service.impl;

import com.learnwy.user.com.learnwy.user.dao.UserRepository;
import com.learnwy.user.modal.UserModal;
import com.learnwy.user.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.util.List;


/**
 * date 2017/9/4
 * create by learnwy
 * lis is GPL-3.0
 */
@Service("userService")
public class UserServiceImpl implements UserService {
    @Autowired
    private UserRepository userRepository;


    @Override
    public List<UserModal> findByUserLoginNameAndUserPassword(String userLoginName, String userPassword) {
        return userRepository.findByUserLoginNameAndUserPassword(userLoginName, userPassword);
    }
}

File: kotlin/_system/src/main/kotlin/com/learnwy/system/user/services/impl/UserServiceImpl.kt
package com.learnwy.system.user.services.impl

import com.learnwy.system.commom.ResquestPage
import com.learnwy.system.user.dto.User
import com.learnwy.system.user.repositories.UserRepository
import com.learnwy.system.user.services.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service

@Service
class UserServiceImpl() : UserService {
    @Autowired
    lateinit var userRepository: UserRepository;

    override fun selectAll(page: ResquestPage): List<User> {
        var pageRequest: PageRequest = PageRequest.of(page.page, page.pageSize);
        var pages: Page<User> = userRepository.findAll(pageRequest);
        page.totalPages = pages.totalPages;
        page.total = pages.numberOfElements;
        return pages.content;
    };
}